pkg/reconciler/instances/istio/action_test.go

```diff
@@ -424,6 +424,142 @@ func newFakeKubeClient() *k8smocks.Client {
+			"string"), mock.AnythingOfType("string"), mock.AnythingOfType("*zap.SugaredLogger")).
+			Return(istioAvailable, nil)
+		performer.On("Uninstall", mock.Anything, mock.AnythingOfType("*zap.SugaredLogger")).Return(nil)
+		action := UninstallAction{performer: &performer}
+
+		// when
+		err := action.Run(actionContext)
+		// then
+		require.NoError(t, err)
+		performer.AssertCalled(t, "Version", mock.Anything, mock.AnythingOfType("string"), mock.
+			AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("*zap.SugaredLogger"))
+		performer.AssertCalled(t, "Uninstall", mock.Anything, mock.AnythingOfType("*zap.SugaredLogger"))
+	})
+	t.Run("should not perform istio uninstall action when istio was not detected on the cluster", func(t *testing.T) {
+		// given
+		factory := workspacemocks.Factory{}
+		provider := chartmocks.Provider{}
+		kubeClient := newFakeKubeClient()
+		actionContext := newFakeServiceContext(&factory, &provider, kubeClient)
+		performer := actionsmocks.IstioPerformer{}
+		performer.On("Version", mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType(
+			"string"), mock.AnythingOfType("string"), mock.AnythingOfType("*zap.SugaredLogger")).
+			Return(noIstioOnTheCluster, nil)
+		action := UninstallAction{performer: &performer}
+
+		// when
+		err := action.Run(actionContext)
+		// then
+		require.NoError(t, err)
+		performer.AssertCalled(t, "Version", mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("*zap.SugaredLogger"))
+		performer.AssertNotCalled(t, "Uninstall", mock.AnythingOfType("string"), mock.AnythingOfType("string"))
+	})
+	t.Run("should not perform istio uninstall action when there is an error detecting istio version", func(t *testing.T) {
+		// given
+		factory := workspacemocks.Factory{}
+		provider := chartmocks.Provider{}
+		kubeClient := newFakeKubeClient()
+		actionContext := newFakeServiceContext(&factory, &provider, kubeClient)
+		performer := actionsmocks.IstioPerformer{}
+		performer.On("Version", mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType(
+			"string"), mock.AnythingOfType("string"), mock.AnythingOfType("*zap.SugaredLogger")).
+			Return(noIstioOnTheCluster, errors.New("error in detecting istio version"))
+		action := UninstallAction{performer: &performer}
+
+		// when
+		err := action.Run(actionContext)
+		// then
+		require.Error(t, err)
+		performer.AssertCalled(t, "Version", mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("*zap.SugaredLogger"))
+		performer.AssertNotCalled(t, "Uninstall", mock.AnythingOfType("string"), mock.AnythingOfType("string"))
+	})
+
+}
+func Test_canUnInstall(t *testing.T) {
+	t.Run("should uninstall when istio is installed", func(t *testing.T) {
+		//given
+		randomVersion := actions.IstioVersion{
```

Comment:
`randomVersion` makes me feel like there is some random factor in the value, whereas it is fixed in the code. We used to call it `fixVersion` or something similar.